### DISCUSSION TITLE
Error on adding obj from API call to FB 'Access to XMLHttpRequest... blocked by CORS policy' API

### DISCUSSION THREAD

**Discussion by jimguistical** [nss-evening-cohort-14/evening-client-side#297] [Capstones!] [answered]
Jim Conner

Issue:
Errors on console screen capped below.  I am working thru the issues related to my patch, for example and creating a getList call to retrieve the array of objects that the user is creating based on the API call in state.  
***What's new to me is the "Access to XMLHttpRequest..." error message at the top.

Of note, I also added my App_Token based on docs here:
https://dev.socrata.com/foundry/data.nashville.gov/797j-5xh2
https://dev.socrata.com/docs/app-tokens.html
So I am wondering if the issue is related to when I was attempting to add my credentials to my calls as well.

Googled:
https://stackoverflow.com/questions/54212220/how-to-fix-access-to-xmlhttprequest-has-been-blocked-by-cors-policy-redirect-i
https://stackoverflow.com/questions/53298478/has-been-blocked-by-cors-policy-response-to-preflight-request-doesn-t-pass-acce
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#preflighted_requests
(I'm working thru this doc now)

https://github.com/jim-conner/capstone-e14/blob/create/src/helpers/data/siteData.js

![Screenshot 2021-06-12 133013](https://user-images.githubusercontent.com/76793747/121785939-670ca100-cb82-11eb-9e4d-157ff52fd4cb.jpg)

**Comment by ajawashington** [comment]
@jim-conner That first resource you posted is helpful. How you tried doing the first answer provided there?

**Comment by TrinityChristiana** [comment, marked as answer]
If you're using your firebase key in the path of a firebase API call, it should always come before `.json`. The only thing that can come after `.json` are queries like `equalTo` and `sortBy`.

[So on this line here](https://github.com/jim-conner/capstone-e14/blob/create/src/helpers/data/siteData.js#L31), `site` should come before `.json`

**Reply by jimguistical** [reply]
Thank you, this worked!  I thought I had replied, but forgot to click send. I needed to work thru the issues in my call, and then the XMLHttpRequest-related warning resolved.
